homeassistant/components/freebox/router.py

```diff
@@ -50,37 +50,41 @@ async def get_api(hass: HomeAssistant, host: str) -> Freepybox:
 class FreeboxRouter:
     """Representation of a Freebox router."""
 
+    _api: Freepybox
+    mac: str
```

Comment:
⚠️ `_api` and `mac` are no longer initialised as None.

Reply:
We need to call setup to set these attributes so not typing them as optional looks not correct. Especially since we have teardown code that should only run if the api is setup.

Reply:
Alternatively I'd do a larger refactor and pass the api to the router instance.

Reply:
I have adjusted this to use `assert` to minimise the code changes.
If it's ok with you, I'll work on a follow-up PR to pass the api to the router instance.